mne/io/egi/egimff.py

```diff
@@ -686,3 +692,200 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+    all_data *= 1e-6  # convert to volts
+
+    # Load metadata into info object
+    # Exclude info['meas_date'] because record time info in
+    # averaged MFF is the time of the averaging, not true record time.
+    ch_names = [channel_naming % (i + 1) for i in
+                range(mff.num_channels['EEG'])]
+    ch_names.extend(egi_info['pns_names'])
+    info = create_info(ch_names, mff.sampling_rates['EEG'], ch_types)
+    info['nchan'] = sum(mff.num_channels.values())
+
+    # Add individual channel info
+    # Get calibration info for EEG channels
+    cals = _get_eeg_calibration_info(fname, egi_info)
+    # Initialize calibration for PNS channels, will be updated later
+    cals = np.concatenate([cals, np.repeat(1, len(egi_info['pns_names']))])
+    ch_coil = FIFF.FIFFV_COIL_EEG
+    ch_kind = FIFF.FIFFV_EEG_CH
+    chs = _create_chs(ch_names, cals, ch_coil, ch_kind, (), (), (), ())
+    chs = _read_locs(fname, chs, egi_info)
+    # Update PNS channel info
+    chs = _add_pns_channel_info(chs, egi_info, ch_names)
+    info['chs'] = chs
+
+    # Add bad channels to info
+    info['description'] = category
+    try:
+        channel_status = categories[category][0]['channelStatus']
+    except KeyError:
+        warn(f'Channel status data not found for condition {category}. '
+             'No channels will be marked as bad.', category=UserWarning)
+        channel_status = None
+    bads = []
+    if channel_status:
+        for entry in channel_status:
+            if entry['exclusion'] == 'badChannels':
+                if entry['signalBin'] == 1:
+                    # Add bad EEG channels
+                    for ch in entry['channels']:
+                        bads.append(channel_naming % ch)
+                elif entry['signalBin'] == 2:
+                    # Add bad PNS channels
+                    for ch in entry['channels']:
+                        bads.append(egi_info['pns_names'][ch - 1])
+    info['bads'] = bads
+
+    # Add EEG reference to info
+    # Initialize 'custom_ref_applied' to False
+    info['custom_ref_applied'] = False
+    with mff.directory.filepointer('history') as fp:
+        history = mffpy.XML.from_file(fp)
+    for entry in history.entries:
+        if entry['method'] == 'Montage Operations Tool':
+            if 'Average Reference' in entry['settings']:
+                # Average reference has been applied
+                projector, info = setup_proj(info)
+            else:
+                # Custom reference has been applied that is not an average
+                info['custom_ref_applied'] = True
+
+    # Get nave from categories.xml
+    try:
+        nave = categories[category][0]['keys']['#seg']['data']
+    except KeyError:
+        warn(f'Number of averaged epochs not found for condition {category}. '
+             'nave will default to 1.', category=UserWarning)
+        nave = 1
+
+    # Let tmin default to 0
+    return EvokedArray(all_data, info, tmin=0., comment=category,
```

Comment:
I wonder if we should add a new `EvokedMFF` class to make it clear that this was read from an MFF file. It seems cleaner than reusing `EvokedArray`:
```
class EvokedMFF(Evoked):

def __init__(self, info, data, ...)
    self.data = data
    self.info = info
    ...
```
But maybe this is more work than it's worth.

Reply:
I think since this `read_evokeds_mff` function resides within this special `egi` library, there won't be much confusion about what file type we are reading from. My vote is that adding a new class would be more trouble than it's worth.